testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py

```diff
@@ -412,3 +425,143 @@ def test_upgrade_testsever_app(params_from_base_suite_setup):
+                                       continuous=True, replicator_authenticator=replicator_authenticator,
+                                       channels=channels)
+    repl = replicator.create(repl_config)
+    log_info("Starting replicator")
+    replicator.start(repl)
+    log_info("Waiting for replicator to go idle")
+    replicator.wait_until_replicator_idle(repl)
+    log_info(replicator.getCompleted(repl), "replicator getCompleted should be 20")
+    replicator.stop(repl)
+
+    sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db)
+
+    # Verify database doc counts
+    cbl_doc_count = db.getCount(cbl_db_obj)
+    cbl_doc_ids = db.getDocIds(cbl_db_obj)
+
+    assert len(sg_docs["rows"]) == 20, "Number of sg docs is not equal to total number of cbl docs and sg docs"
+    assert cbl_doc_count == 20, "Did not get expected number of cbl docs"
+
+    # Check that all doc ids in SG are also present in CBL
+    sg_ids = [row["id"] for row in sg_docs["rows"]]
+    for doc in cbl_doc_ids:
+        assert doc in sg_ids
+    prebuilt_db_path = db.getPath(cbl_db_obj)
+    log_info("prebuilt_db_path", prebuilt_db_path)
+
+    testserver2 = TestServerFactory.create(platform=second_liteserv_platform,
+                                           version_build=second_liteserv_version,
+                                           host=second_liteserv_host,
+                                           port=liteserv_port,
+                                           community_enabled=community_enabled,
+                                           debug_mode=debug_mode)
+    log_info("Downloading TestServer ...")
+    # Download TestServer app
+    testserver2.download()
+
+    log_info("Installing TestServer ...")
+    # Install TestServer app
+    if device_enabled:
+        testserver2.install_device()
+        testserver2.start_device("{}/logs/{}-{}-{}.txt".format(RESULTS_DIR, type(testserver2).__name__,
+                                                               test_name_cp,
+                                                               datetime.datetime.now()))
+    else:
+        testserver2.install()
+        testserver2.start("{}/logs/{}-{}-{}.txt".format(RESULTS_DIR, type(testserver2).__name__,
+                                                        test_name_cp,
+                                                        datetime.datetime.now()))
+
+    db = Database(base_url)
+    temp_db_name = "temp_db_name"
+    cbl2_db_obj = db.create(temp_db_name)
+
+    new_db_path = db.getPath(cbl2_db_obj)
+    db.deleteDB(cbl2_db_obj)
+    assert "Copied" == utils_obj.copy_files(prebuilt_db_path, new_db_path)
+    new_cbl_db_obj = db.create(cbl_db_name)
+
+    authenticator = Authenticator(base_url)
+    replicator_authenticator = authenticator.authentication(session, cookie, authentication_type="session")
+    repl_config = replicator.configure(new_cbl_db_obj, target_url=sg_blip_url, replication_type="pull",
```

Comment:
can you use configure_and_replicate which covers belows steps too?

Reply:
done